### PR TITLE
refactor: implement rest API endpoint for get app and get apps

### PR DESCRIPTION
**Pull request by ridvanakca**

### CODE REVIEW

**Review by barinali — CHANGES_REQUESTED** [review]

**Review by barinali — APPROVED** [review]

File: packages/web/src/hooks/useActionSubsteps.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { useQuery } from '@tanstack/react-query';
+
+import api from 'helpers/api';
+
+export default function useActionSubsteps(appKey, actionKey) {
+  const query = useQuery({
+    queryKey: ['actionSubsteps', appKey, actionKey],
+    queryFn: async ({ payload, signal }) => {
```
Comment (kattoczko) [inline packages/web/src/hooks/useActionSubsteps.js:8]: payload is not used and could be removed

File: packages/web/src/hooks/useActions.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { useQuery } from '@tanstack/react-query';
+
+import api from 'helpers/api';
+
+export default function useActions(appKey) {
+  const query = useQuery({
+    queryKey: ['actions', appKey],
+    queryFn: async ({ payload, signal }) => {
```
Comment (kattoczko) [inline packages/web/src/hooks/useActions.js:8]: payload is not used and could be removed

File: packages/web/src/hooks/useAdminAppAuthClient.ee.js [hunk 3]
```diff
@@ -6,7 +6,9 @@ export default function useAdminAppAuthClient(id) {
   const query = useQuery({
     queryKey: ['adminAppAuthClient', id],
     queryFn: async ({ payload, signal }) => {
```
Comment (kattoczko) [inline packages/web/src/hooks/useAdminAppAuthClient.ee.js:8]: payload is not used and could be removed

File: packages/web/src/hooks/useApp.js [hunk 4]
```diff
@@ -1,12 +1,19 @@
...
-  const { data, loading } = useQuery(GET_APP, {
-    variables: { key },
+import { useQuery } from '@tanstack/react-query';
+
+import api from 'helpers/api';
+
+export default function useApp(appKey) {
+  const query = useQuery({
+    queryKey: ['app', appKey],
+    queryFn: async ({ payload, signal }) => {
```
Comment (kattoczko) [inline packages/web/src/hooks/useApp.js:8]: payload is not used and could be removed

File: packages/web/src/hooks/useAppAuth.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import { useQuery } from '@tanstack/react-query';
+
+import api from 'helpers/api';
+
+export default function useAppAuth(appKey) {
+  const query = useQuery({
+    queryKey: ['appAuth', appKey],
+    queryFn: async ({ payload, signal }) => {
```
Comment (kattoczko) [inline packages/web/src/hooks/useAppAuth.js:8]: payload is not used and could be removed

File: packages/web/src/hooks/useApps.js [hunk 6]
```diff
@@ -1,12 +1,19 @@
...
+import { useQuery } from '@tanstack/react-query';
+
+import api from 'helpers/api';
+
 export default function useApps(variables) {
-  const { data, loading } = useQuery(GET_APPS, {
-    variables,
+  const query = useQuery({
+    queryKey: ['apps', variables],
+    queryFn: async ({ payload, signal }) => {
```
Comment (kattoczko) [inline packages/web/src/hooks/useApps.js:8]: payload is not used and could be removed

File: packages/web/src/hooks/useLazyApps.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+  React.useEffect(() => {
+    abortControllerRef.current = new AbortController();
+
+    return () => {
+      abortControllerRef.current?.abort();
+    };
+  }, [appName]);
+
+  const query = useMutation({
+    mutationFn: async ({ payload }) => {
```
Comment (kattoczko) [inline packages/web/src/hooks/useLazyApps.js:18]: payload is not used and could be removed

File: packages/web/src/hooks/useTriggerSubsteps.js [hunk 8]
```diff
@@ -0,0 +1,22 @@
+import { useQuery } from '@tanstack/react-query';
+
+import api from 'helpers/api';
+
+export default function useTriggerSubsteps(appKey, triggerKey) {
+  const query = useQuery({
+    queryKey: ['triggerSubsteps', appKey, triggerKey],
+    queryFn: async ({ payload, signal }) => {
```
Comment (kattoczko) [inline packages/web/src/hooks/useTriggerSubsteps.js:8]: payload is not used and could be removed

File: packages/web/src/hooks/useTriggers.js [hunk 9]
```diff
@@ -0,0 +1,19 @@
+import { useQuery } from '@tanstack/react-query';
+
+import api from 'helpers/api';
+
+export default function useTriggers(appKey) {
+  const query = useQuery({
+    queryKey: ['triggers', appKey],
+    queryFn: async ({ payload, signal }) => {
```
Comment (kattoczko) [inline packages/web/src/hooks/useTriggers.js:8]: payload is not used and could be removed

**Pull request merged**